src/main/webapp/discussion.js

```diff
@@ -103,53 +129,77 @@ async function fetchDiscussion() {
+    this.shadowRoot.appendChild(shadow);
+
+    const username = comment.author.email.split('@')[0];
+    let timestampPrefix = '';
+    if (!comment.parentKey.value) {
+      // Only display timestamp on root comments.
+      timestampPrefix = `${this.timestampToString(comment.timestamp)} - `;
+    }
+    const header = `${timestampPrefix}${username} on ${comment.created}`;
+    this.setSlotSpan(SLOT_HEADER, header);
 
-  const content = document.createElement('span');
-  content.innerText = `${comment.author.email} says ${comment.content}`;
-  element.appendChild(content);
+    this.setSlotSpan(SLOT_CONTENT, comment.content);
 
-  const repliesDiv = document.createElement('div');
+    this.appendChild(this.createReplies(comment.replies));
 
-  const repliesList = document.createElement('ul');
-  for (const reply of comment.replies) {
-    repliesList.appendChild(createComment(reply));
+    const replyForm = this.shadowRoot.querySelector(SELECTOR_REPLY_FORM);
+
+    this.shadowRoot.querySelector(SELECTOR_SHOW_REPLY).onclick = () => {
+      $(replyForm).collapse('show');
+    };
+    this.shadowRoot.querySelector(SELECTOR_CANCEL_REPLY).onclick = () => {
+      $(replyForm).collapse('hide');
+    };
+    this.shadowRoot.querySelector(SELECTOR_POST_REPLY).onclick = () => {
+      const textarea = this.shadowRoot.querySelector(SELECTOR_REPLY_TEXTAREA);
+      postReply(textarea, comment.commentKey.id);
+    };
   }
-  repliesDiv.appendChild(repliesList);
-
-  const replyButton = document.createElement('button');
-  replyButton.innerText = 'Reply';
-  element.appendChild(replyButton);
-  replyButton.onclick = () => {
-    createReplySubmission(repliesDiv);
-    replyButton.remove();
-  };
-  element.appendChild(repliesDiv);
-
-  return element;
-}
 
-/**
- * Creates a reply textarea and submit button within {@code repliesDiv}.
- *
- * <p>The parent of {@code repliesDiv} should be a comment element created by
- * {@code createComment}.
- */
-function createReplySubmission(repliesDiv) {
-  const div = document.createElement('div');
-  const textarea = document.createElement('textarea');
-  const submit = document.createElement('button');
-  const parentId = repliesDiv.parentElement.getAttribute(ATTR_ID);
-  submit.innerText = 'Post';
-  submit.onclick = () => {
-    postAndReload(textarea, parentId);
-  };
-  div.appendChild(textarea);
-  div.appendChild(submit);
-
-  repliesDiv.prepend(div);
+  /**
+   * Coverts a Date representing a video timestamp into a string.
+   */
+  // TODO: Implement using a utility function.
+  timestampToString(timestamp) {
+    const ts = new Date(timestamp);
+    const seconds = ts.getSeconds().toString().padStart(2, '0');
+    const minutes = ts.getMinutes().toString().padStart(2, '0');
+    return `${minutes}:${seconds}`;
+  }
+
+  /**
+   * Creates a Discussion Comment for every reply to this comment, and adds
+   * them to a {@code div} in the replies slot.
```

Comment:
What is "replies slot"

Reply:
Templates have a number of named slots that can be filled with other elements.  I will change some things around to make it more clear